frappe/public/js/frappe/desk.js

```diff
@@ -54,13 +54,19 @@ frappe.Application = class Application {
 		this.setup_copy_doc_listener();
 
 		frappe.ui.keys.setup();
-
+		
 		frappe.ui.keys.add_shortcut({
 			shortcut: 'shift+ctrl+g',
 			description: __('Switch Theme'),
 			action: () => {
-				frappe.theme_switcher = new frappe.ui.ThemeSwitcher();
-				frappe.theme_switcher.show();
+				if (cur_dialog) {
+					frappe.theme_switcher.hide();
+				} else {
+					if (!frappe.theme_switcher) {
+						frappe.theme_switcher = new frappe.ui.ThemeSwitcher();
+					}
+					frappe.theme_switcher.show();
+				}
```

Comment:
```suggestion
				if (frappe.theme_switcher && frappe.theme_switcher.dialog.is_visible) {
					frappe.theme_switcher.hide();
				} else {
					frappe.theme_switcher = new frappe.ui.ThemeSwitcher();
					frappe.theme_switcher.show();
				}
```

Only, hide if theme_switcher dialog is visible not cur_dialog
And always create a new frappe.theme_switcher when we want to show because if theme switcher is already created and if another dialog is opened after that and then again you open theme switcher dialog it will go behind the currently opened dialog.